3.cellprofiling/scripts/intensity.py

```diff
@@ -172,29 +95,43 @@ def process_combination(
 # In[ ]:
 
 
-# Generate all combinations of compartments and channels
-combinations = list(
-    product(image_set_loader.compartments, image_set_loader.image_names)
+object_loader = ObjectLoader(
+    image_set_loader.image_set_dict[channel],
+    image_set_loader.image_set_dict[compartment],
+    channel,
+    compartment,
 )
-cores = multiprocessing.cpu_count()
-print(f"Using {cores} cores for processing.")
-# Use multiprocessing to process combinations in parallel
-with multiprocessing.Pool(processes=cores) as pool:
-    results = list(
-        tqdm(
-            pool.imap(
-                partial(
-                    process_combination,
-                    image_set_loader=image_set_loader,
-                    output_parent_path=output_parent_path,
-                ),
-                combinations,
-            ),
-            desc="Processing combinations",
-        )
+if processor_type == "GPU":
+    output_dict = measure_3D_intensity_gpu(object_loader)
+elif processor_type == "CPU":
+    output_dict = measure_3D_intensity_CPU(object_loader)
```

Comment:
Would it make sense for some of these CPU/GPU decisions to be made within the context of the function? This might collapse the need for logic here and make it so you can pass `processor_type` directly to the function. I say this without knowing what would go into this - so it could be that the answer is: "no, it'd be too complex!"